src/services/localStorage.tsx

```diff
@@ -0,0 +1,38 @@
+import {Dispatch} from "react";
```

Comment:
Looks weird that in this file you requires a `React` dependency.
I think that is only for typing, but `(teamType: TeamType) => void` should works too 🤔. It deep what you are passing is a callback function.